x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/MetadataAttribute.java

```diff
@@ -99,4 +141,18 @@ public static DataType dataType(String name) {
     public static boolean isSupported(String name) {
         return ATTRIBUTES_MAP.containsKey(name);
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (false == super.equals(obj)) {
+            return false;
+        }
+        MetadataAttribute other = (MetadataAttribute) obj;
+        return searchable == other.searchable;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), searchable);
+    }
```

Comment:
Adding this to the equality doesn't seem to have broken anything and feels more correct.